### PR TITLE
Final Independent Assessment - Michael Clampett 1810 BEE - Reunion

### PR DESCRIPTION
**Pull request by SyntheticAutomation**
Final Independent Assessment - Michael Clampett 1810 BEE - Reunion

### CODE REVIEW

**Comment by SyntheticAutomation** [comment]
Completed everything up to the detailed_breakout method in Iteration 4. Almost had it!

File: lib/reunion.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      activity.total_cost
+    end
+  end
+
+  def breakout
+    final_output = {}
+    @activities.each do |activity|
+      activity.owed.each do |name, money_spent|
+        it_exists = final_output.include?(name)
+        it_exists ? final_output[name] += money_spent : final_output[name] = money_spent
```
Comment (s-espinosa) [inline lib/reunion.rb:25]: Could avoid this ternary if you created `final_output` using Hash.new(0).

File: lib/reunion.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    end
+    final_output
+  end
+
+  def summary
+    summary_output = ""
+    breakout.each do |name, balance|
+      summary_output << "#{name}: #{balance}\n"
+    end
+    summary_output.chomp
```
Comment (s-espinosa) [inline lib/reunion.rb:36]: 👍

File: test/reunion_test.rb [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    { activity: "Drinks", payees: ["Maria", "Luther"], amount: 20},
+    { activity: "Bowling", payees: ["Maria", "Luther"], amount: -10},
+    { activity: "Jet Skiing", payees: ["Maria", "Luther"], amount: -10}],
+  "Nemo" => [
+    { activity: "Jet Skiing", payees: ["Maria", "Luther"], amount: -10}]
+  }
+    assert_equal expected, reunion.detailed_breakout
+  end
+
+end
```
Comment (s-espinosa) [inline test/reunion_test.rb:116]: Tests look good!

**Review by s-espinosa — COMMENTED** [review]
@SyntheticAutomation Looks like you got through I3, and made some good progress on I4, which meets our requirements. Small comment below. Let us know if you have any questions. Thanks!

**Pull request closed without merging**